### ISSUE TITLE
Studiomdl can't compile f_anm and z_anm

### ISSUE THREAD

**Issue by KingDavidW** [robotboy655/gmod-animations#2]
GMod's Studiomdl can not compile the f_anm and z_anm without corruption to the .ani files. Tested this with at least a dozen people running either Windows 7 or 10. Output will result in garbled text, and the ingame result will be the FPS tanking whenever a model that inherits those .ani files is used.

F_anm problem can be repaired by changing the female commonbones finger bone definition

Before:
```$definebone "ValveBiped.Bip01_L_Finger4" "ValveBiped.Bip01_L_Hand" 3.549243 0.198824 -1.489468 21.265158 0.597196 -12.141549 -0.000000 -0.000001 -0.000000 -0.000001 0.000002 0.000000```

After: (Change 21 to 20) 
```$definebone "ValveBiped.Bip01_L_Finger4" "ValveBiped.Bip01_L_Hand" 3.549243 0.198824 -1.489468 20.265158 0.597196 -12.141549 -0.000000 -0.000001 -0.000000 -0.000001 0.000002 0.000000```

I'm finding that corruption happens when # of blocks is over 256. m_anm has 256, pre-repaired f_anm will have 257 and z_anm will have 258. The above change to the commonbones makes f_anm have 256.

Screenshot of the .ani garbling of f_anm:
https://cdn.discordapp.com/attachments/589120351238225940/768977921514799113/unknown.png

**Comment by KingDavidW** [comment]
I can confirm this is definitely a MEMORY LIMIT with studiomdl. Directely related to the # of blocks created. Simple way to test:

```
// Optimization commands
$animblocksize 32
```
Change to:
```
// Optimization commands
$animblocksize 36
```

Blocks per .ani decrease, no corruption during final compile, animation runs fine.

**Comment by robotboy655** [comment]
This is hitting the hard limit of 256 anim blocks. I am surprised it doesn't just outright crash the compiler.

Looked at TF2s and CS:GOs compiler sources, they seem to also just not handle this at all.
In GMod, I ended up just canceling the compilation if we are about to try to write to out of bounds memory.
I also increased the limit to 512 as it seems that this limit only affects the compiler itself.
